### PR TITLE
[Ready]Ghosts can now Health Scan

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Ghosts now have a Toggle Health Scan verb, which allows them to health scan mobs they click on.
/:cl:

Gives some nice observer information, like damage types, time of death, diseases, and other afflictions that aren't evident without a full health scan.

### CODE REVIEW

**Comment by FantasticFwoosh** [comment]
Well the medhud verbs were a bit of a waste of time then.

**Comment by Shadowlight213** [comment]
Could just tie it into the data hud toggle instead of adding a new verb for this. Like how it works for examining a blob.

**Comment by Bawhoppen** [comment]
@Shadowlight213 I feel like people'd want to be able to see that stuff but not get a giant wall of text every time I examine someone

**Comment by Shadowlight213** [comment]
True.

**Comment by RandomMarine** [comment]
I wish ghosts had an atmos scan but to my knowledge there's no generalized proc like that.

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -675,6 +676,18 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+	set name = "Toggle Health Scan"
+	set desc = "Toggles whether you health-scan living beings on click"
+	set category = "Ghost"
+
+	if(health_scan) //remove old huds
+		to_chat(src, "<span class='notice'>Health scan disabled.</span>")
+		health_scan = 0
+	else
+		to_chat(src, "<span class='notice'>Health scan enabled.</span>")
+		health_scan = 1
```
Comment (Cyberboss) [inline code/modules/mob/dead/observer/observer.dm]: TRUE

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -675,6 +676,18 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
 		data_huds_on = 1
 
+/mob/dead/observer/verb/toggle_health_scan()
+	set name = "Toggle Health Scan"
+	set desc = "Toggles whether you health-scan living beings on click"
+	set category = "Ghost"
+
+	if(health_scan) //remove old huds
+		to_chat(src, "<span class='notice'>Health scan disabled.</span>")
+		health_scan = 0
```
Comment (Cyberboss) [inline code/modules/mob/dead/observer/observer.dm]: FALSE

File: code/_onclick/observer.dm [hunk 3]
```diff
@@ -53,7 +53,9 @@
 	if(user.client)
 		if(IsAdminGhost(user))
 			attack_ai(user)
-		if(user.client.prefs.inquisitive_ghost)
+		else if(isliving(src) && user.health_scan)
```
Comment (Cyberboss) [inline code/_onclick/observer.dm]: override mob/living/atttack_ghost instead

File: code/game/objects/items/devices/scanners.dm [hunk 4]
```diff
@@ -106,8 +106,8 @@ MASS SPECTROMETER
 
 
 // Used by the PDA medical scanner too
-/proc/healthscan(mob/living/user, mob/living/M, mode = 1, advanced = FALSE)
```
Comment (Cyberboss) [inline code/game/objects/items/devices/scanners.dm]: has this really changed?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/scanners.dm (hunk 4 above)
Comment (XDTM) [reply code/game/objects/items/devices/scanners.dm]: From mob/living/user to mob/user

**Comment by The-clown-known-as-Pepe** [comment]
can we get chem scan too?

**Comment by AnonymousNow** [comment]
This sounds like the start of a GhostHUD.

File: code/_onclick/observer.dm [hunk 5]
```diff
@@ -53,9 +53,16 @@
 	if(user.client)
 		if(IsAdminGhost(user))
 			attack_ai(user)
-		if(user.client.prefs.inquisitive_ghost)
+		else if(user.client.prefs.inquisitive_ghost)
 			user.examinate(src)
 
+/mob/living/attack_ghost(mob/dead/observer/user)
+	if(user.client)
+		if(user.health_scan)
```
Comment (Cyberboss) [inline code/_onclick/observer.dm]: Combine these ifs

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
There's not a lot of appetite among the maintainers for this FYI

**Comment by XDTM** [comment]
My bad, i suppose, i left that review hanging

**Review by Cyberboss — APPROVED** [review]

**Comment by optimumtact** [comment]
If nobody is going to merge this I'm going to close it.

**Comment by AnonymousNow** [comment]
Shame, for a popular PR.

**Comment by Cyberboss** [comment]
Honestly I forgot about it, I'll get it when I get back

**Comment by optimumtact** [comment]
don't lie to me cyberboss

**Pull request merged**